File: orbit.app/interfaces/api/meal-planner/AddRecipeRequest.ts
export interface RecipeIngredient {
  name: string
  quantity: string | null
}

export interface RecipeStep {
  stepNumber: number
  instruction: string
}

export interface AddRecipeRequest {
  name: string
  description: string
  ingredients: RecipeIngredient[] | null
  steps: RecipeStep[] | null
  prepTimeMinutes: number | null
  cookTimeMinutes: number | null
  servings: number | null
}

File: Orbit.Domain/DTOs/MealPlanner/AddRecipeRequest.cs
namespace Orbit.Domain.DTOs.MealPlanner
{
    public class AddRecipeRequest
    {
        public required string Name { get; set; }
        public required string Description { get; set; }
        public RecipeIngredientDto[]? Ingredients { get; set; }
        public RecipeStepDto[]? Steps { get; set; }
        public int? PrepTimeMinutes { get; set; }
        public int? CookTimeMinutes { get; set; }
        public int? Servings { get; set; }
    }
}
